include/experimental/__p0009_bits/mdspan.hpp

```diff
@@ -307,13 +307,38 @@ class mdspan
 };
 
 #if defined(_MDSPAN_USE_CLASS_TEMPLATE_ARGUMENT_DEDUCTION)
+#ifdef _MDSPAN_USE_P2554
+MDSPAN_TEMPLATE_REQUIRES(
+  class ElementType, class... SizeTypes,
+  /* requires */ _MDSPAN_FOLD_AND(_MDSPAN_TRAIT(is_integral, SizeTypes) /* && ... */) &&
+  (sizeof...(SizeTypes) > 0)
+)
+mdspan(ElementType*, SizeTypes...)
+  -> mdspan<ElementType, ::std::experimental::dextents<sizeof...(SizeTypes)>>;
+
+MDSPAN_TEMPLATE_REQUIRES(
+  class Pointer,
+  (!_MDSPAN_TRAIT(is_array, Pointer))
+)
+mdspan(Pointer&) -> mdspan<std::remove_pointer_t<Pointer>, extents<>>;
```

Comment:
ok. Funnily enough I found a bug in the proposal through me writing the test slightly differently then what I had tested before. The deduction guide for this guy needs to take const Pointer& instead of Pointer& otherwise mdspan a(d.data()) doesn't work